include/CPUCopy.hpp

```diff
@@ -22,19 +22,14 @@ void copy(fg::Image& out, const T * dataPtr)
 }
 
 /*
- * Below functions takes any renderable forge object that has following member functions
- * defined
- *
- * `unsigned Renderable::vbo() const;`
- * `unsigned Renderable::size() const;`
- *
- * Currently fg::Plot, fg::Histogram objects in Forge library fit the bill
+ * Below functions expects OpenGL resource Id and size in bytes to copy the data from
+ * cpu memory location to graphics memory
  */
-template<class Renderable, typename T>
-void copy(Renderable& out, const T * dataPtr)
+template<typename T>
+void copy(const int resourceId, const size_t resourceSize, const T * dataPtr)
```

Comment:
I am not sure why this needs to be a template (unless you are asserting the data type to be the same as the one inside `resourceId`). If not it should be removed from all the places and use `void *` instead.